### ISSUE TITLE
Add a duplicate warning when creating and editing aggregate reports [3]

### ISSUE THREAD

**Issue by bernardsilenou** [hzi-braunschweig/SORMAS-Project#9039]
### Feature Description

Currently the web app and mobile app allow users to create additional aggregate reports for a jurisdiction entity, without checking if other users have already done so.

### Proposed Change

- [x] When the user edits/creates an aggregate report retrieve all existing reports for the same jurisdiction and epi week, created by other users.
- [x] If any exist display a warning between the filters/jurisdiction selection and the data entry part: "There already exist aggregate reports for 'district/facility name' in this epi week. The corresponding disease names are displayed in red."
- [x] For each disease that already has an entry display the disease label in red.
- [x] Same for the mobile app

### Implementation Details

Blocked by #9119

### Additional Information

1. If multiple reports are created in a week, the reports are considered as independent and sum together. This leads to overcounting -> addressed by #9068
2. Add a duplicate detection method to identify reports with the same jurisdiction and epi week -> #9073
3. It is possible that a duplicate report can be created when using the mobile app offline. During synch, use the most recently modified report as the right one while the others are stored as duplicates. It may not be good to delete them but instead add a filter (on the web app) like merger duplicate to allow users to identify duplicates + decides which to keep as the right record + option to delete the other pairs. -> #9068 and #9073
4. eg if there are two SOs' in district A, if SO1 submit a report for week1 2022, and SO2/ SO1 try to create another report for the same week, then show a pop up that a report exists for that week and ask the user to edit it if needed.  However, a hospital informant can create a report for the same week in district A since this is another user role. A second or the same informant should not also be able to create a new report in the same week.

Caution: A duplicate check is already partly implemented, see `AggregateReportsEditLayout.checkForExistingData` in 2020-03 with #1526.

**Comment by MartinWahnschaffe** [comment]
@bernardsilenou 
I have changed the issue description and moved a big part of your points to the additional information section, because they are addressed by other issues I have created.

Two additional notes:

On 3: The duplicate detection will not happen during sync. Instead both entries will stay side-by-side in the database without having marked one as a duplicate. #9068 will take care of ignoring those duplicates when aggregating. #9073 will allow the user to identify and delete duplicates.

On 4: I don't fully agree with this. What if two informants in a health facility share the task to report, each one taking care of a subset of the disease. I guess there may be other examples like this. Based on that I would not forbid users to create a report for their own jurisdiction, but I'd display warnings like describe above.

Important: We will only have a real duplicate here, when two users report non-0-numbers for the exact same jurisdiction and disease within an epi week.

**Comment by bernardsilenou** [comment]
@MartinWahnschaffe 
Your solution for point 4 is much better

**Comment by MartinWahnschaffe** [comment]
@bernardsilenou @MateStrysewske Do you think we should show the warning for the whole form or should we do individual warnings for every single diease + age group within the form? For now I would go with just a warning for the whole form.

**Comment by MateStrysewske** [comment]
@MartinWahnschaffe I'm in favour of individual warnings because I don't really see a general warning being of much use for the user. What's the expected action to be taken by them if this warning pops up?

**Comment by MartinWahnschaffe** [comment]
Agreed. I have updated the description accordingly.

**Comment by adinaflorea9** [comment]
Currently, there is a duplicate warning pop-up when aggregate reports are created in the web-app (vaadin-app) only.

The current behavior is the following, and it should be looked over and fixed when this ticket is implemented.

1. When creating new aggregated reports, if the epi week and jurisdiction (up to facility/point of entry) match an existing report, a 'duplicate detection' pop-up is triggered. 

I noticed that this pop-up is triggered only if the jurisdiction matches up to the facility/point of entry selected. Since reports can be created without a facility or point of entry, when checkbox 2 of the proposed change is implemented, the current behavior will be fixed (i.e. the duplicate detection will be triggered when the jurisdiction matches up to district only as well).

2. **Hospital Informant User** - the duplicate detection pop-up for aggregated reports is triggered for this user 'behind' the 'Create new aggregated report' form - so the user first creates a full report, saves it, and only afterwards they can see the duplicate pop-up; regardless of the selection they make in the pop-up, the new report is saved already. 

Since the pop-up is triggered when the epi week and jurisdiction match up to facility/point of entry, and for this user all of the jurisdiction fields are read only and prefilled, the pop-up is always triggered.

Here the pop-up should be displayed in front of the 'Create new aggregate report' form, as it is for the other users.

**Comment by adinaflorea9** [comment]
Verified ticket on the local machine using the latest version of SORMAS from the development branch - 1.75.0-SNAPSHOT(81bdbe6).

#10024 - change request to also display age groups in red is reports already exist for them.
